katharsis-core/src/main/java/io/katharsis/resource/registry/responseRepository/RelationshipRepositoryAdapter.java

```diff
@@ -17,76 +28,106 @@
+                .removeRelations(source, targetIds, fieldName, toQueryParams(queryAdapter));
+        } else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
+        	((QuerySpecRelationshipRepository)relationshipRepository).removeRelations(source, targetIds, fieldName);
         } else {
             ((RelationshipRepository) relationshipRepository).removeRelations(source, targetIds, fieldName);
         }
         return new JsonApiResponse();
     }
 
-    public JsonApiResponse findOneTarget(T_ID sourceId, String fieldName, QueryParams queryParams) {
+    public JsonApiResponse findOneTarget(T_ID sourceId, String fieldName, QueryAdapter queryAdapter) {
         Object resource;
         if (isAnnotated) {
             resource = ((AnnotatedRelationshipRepositoryAdapter) relationshipRepository)
-                .findOneTarget(sourceId, fieldName, queryParams);
+                .findOneTarget(sourceId, fieldName, toQueryParams(queryAdapter));
+        } else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
+        	resource = ((QuerySpecRelationshipRepository)relationshipRepository).findOneTarget(sourceId, fieldName, toQuerySpec(queryAdapter));
         } else {
             resource = ((RelationshipRepository) relationshipRepository)
-                .findOneTarget(sourceId, fieldName, queryParams);
+                .findOneTarget(sourceId, fieldName, toQueryParams(queryAdapter));
         }
-        return getResponse(relationshipRepository, resource, queryParams);
+        return getResponse(relationshipRepository, resource, queryAdapter);
     }
 
-    public JsonApiResponse findManyTargets(T_ID sourceId, String fieldName, QueryParams queryParams) {
+	public JsonApiResponse findManyTargets(T_ID sourceId, String fieldName, QueryAdapter queryAdapter) {
         Object resources;
         if (isAnnotated) {
             resources = ((AnnotatedRelationshipRepositoryAdapter) relationshipRepository)
-                .findManyTargets(sourceId, fieldName, queryParams);
+                .findManyTargets(sourceId, fieldName, toQueryParams(queryAdapter));
+        }else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
+            	resources = ((QuerySpecRelationshipRepository)relationshipRepository).findManyTargets(sourceId, fieldName, toQuerySpec(queryAdapter));
         } else {
             resources = ((RelationshipRepository) relationshipRepository)
-                .findManyTargets(sourceId, fieldName, queryParams);
+                .findManyTargets(sourceId, fieldName, toQueryParams(queryAdapter));
         }
-        return getResponse(relationshipRepository, resources, queryParams);
+        return getResponse(relationshipRepository, resources, queryAdapter);
     }
+	
 
 	public Object getRelationshipRepository() {
 		return relationshipRepository;
 	}
+	
+	public FilterOperator getDefaultOperator() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1161)